src/api/tests/test_class.py

```diff
@@ -0,0 +1,21 @@
+from .util import Client
+
+def test_success(upload, client: Client):
+    #Make sure the upload works
+    assert upload.status_code == 200
+    r = client.get("/api/class?semester=SPRING%202020&search=NUMERICAL%20COMPUTING")
```

Comment:
I believe your problem with using params earlier was you had the url encoded strings instead of the original strings.

Try using params with params = {'semester': 'SPRING 2020', 'search': 'NUMERICAL COMPUTING'}

This should work, and you should follow this for the rest of your API calls as well